bundles/org.openhab.ui.homebuilder/web/src/restItems.js

```diff
@@ -0,0 +1,213 @@
+ */
+export function addObjectGroups(model) {
+    let items = [];
+    let chosenObjects = getChosenObjects(model);
+
+    chosenObjects.forEach(function(dev) {
+        let object = _.find(objects, { value: dev });
+
+        if (object) {
+            let type = object.type.split(':');
+            let groupType = _.first(type);
+            let groupFuncName = type[1] ? type[1].split('(')[0] : '';
+            let groupFuncArgs = type[1] && type[1].split('(')[1] ? type[1].split('(')[1]
+                .split(',')
+                .map((arg) => arg.trim().replace(/\W/g, '')) : [];
+
+            let item = {
+                type: 'Group',
+                name: GROUP_PREFIX + object.value,
+                label: object.name || object.value,
+                category: model.itemsIcons ? object.icon : '',
+                groupNames: ['Home'],
+                groupType: groupType,
+                entryType: 'objectGroup'
+            };
+
+            if (groupFuncName) {
+                item = _.extend({}, item, {
+                    function: {
+                        name: groupFuncName
+                    }
+                });
+
+                if (!_.isEmpty(groupFuncArgs)) {
+                    item.function.params = groupFuncArgs;
+                }
+            }
+
+            items.push(item);
+        }
+    });
+
+    return _.uniq(items);
+}
+
+/**
+ * Gets list of objects chosen
+ * from collection
+ * 
+ * @param {*} model 
+ * @return {Array}
+ */
+export function getChosenObjects(model) {
+    return _.chain(model)
+        .pickBy((value, key) => _.endsWith(key, OBJECTS_SUFFIX))
+        .flatMap()
+        .map((item) => item.value)
+        .uniq()
+        .value() || [];
+}
+
+/**
+ * For a given object it creates a HomeKit-compatible
+ * set of tags.
+ * 
+ * @param {Object} object 
+ * @param {Object} model 
+ * @return {Array}
+ */
+function addTags(object, model) {
+    var type = _.first(object.type.split(':'));
+    var tags = [];
+
+    switch (type) {
+        case 'Switch':
+        case 'Dimmer':
+        case 'Color':
+            tags.push('Switchable');
+            break;
+        default:
+            break;
+    }
+
+    switch (object.value) {
+        case 'Lamp':
+        case 'Light':
+            tags = ['Lighting'];
+            break;
+        case 'Motion':
+            tags = [];
+            break;
+        default:
+            break;
+    }
+
+    return model.itemsTags ? tags : [];
+}
+
+export function getItems(model) {
+    let items = [{
+        type: 'Group',
+        name: 'Home',
+        label: model.homeName,
+        category: model.itemsIcons ? 'house' : '',
+        entryType: 'home'
+    }];
+
+    for (var i = 0; i < model.floorsCount; i++) {
+        var floor = floors[i];
+
+        items = [
+            ...items,
+            ...addFloors(floor, model),
+            ...addRooms(floor, model)
+        ];
+    }
+
+    items = [
+        ...items,
```

Comment:
What does this syntax do?

Reply:
It's JavaScript's [spread operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_operator).

Reply:
Goal achieved I learned something today 😉